corehq/messaging/scheduling/scheduling_partitioned/models.py

```diff
@@ -679,3 +679,13 @@ class Meta(AbstractTimedScheduleInstance.Meta):
         unique_together = (
             ('case_id', 'timed_schedule_id', 'recipient_type', 'recipient_id'),
         )
+
+    def additional_deactivation_condition_reached(self):
+        from corehq.messaging.tasks import get_cached_rule
```

Comment:
Does this cause circular import as well, if on top of the file?

Reply:
yeah, this import also causes a circular import as the `tasks` file imports the `AutomaticUpdateRule` class.